kms/azurekms/utils.go

```diff
@@ -101,23 +104,78 @@ func parseKeyName(rawURI string, defaults defaultOptions) (vault, name, version
 }
 
 func convertKey(key *azkeys.JSONWebKey) (crypto.PublicKey, error) {
-	// Hack to be able to properly convert keys in HSM.
-	if key != nil && key.Kty != nil {
-		switch *key.Kty {
-		case azkeys.JSONWebKeyTypeECHSM:
-			key.Kty = pointer(azkeys.JSONWebKeyTypeEC)
-		case azkeys.JSONWebKeyTypeRSAHSM:
-			key.Kty = pointer(azkeys.JSONWebKeyTypeRSA)
-		}
+	if key == nil || key.Kty == nil {
+		return nil, errors.New("invalid key: missing kty value")
 	}
 
-	b, err := json.Marshal(key)
-	if err != nil {
-		return nil, errors.Wrap(err, "error marshaling key")
+	switch *key.Kty {
+	case azkeys.JSONWebKeyTypeEC, azkeys.JSONWebKeyTypeECHSM:
+		return ecPublicKey(key.Crv, key.X, key.Y)
+	case azkeys.JSONWebKeyTypeRSA, azkeys.JSONWebKeyTypeRSAHSM:
+		return rsaPublicKey(key.N, key.E)
+	case azkeys.JSONWebKeyTypeOct, azkeys.JSONWebKeyTypeOctHSM:
+		return octPublicKey(key.K)
+	default:
+		return nil, fmt.Errorf("invalid key: unsupported kty '%s'", *key.Kty)
```

Comment:
Would `%q` work for this one?

Reply:
I'll change this. I was keeping the format go-jose was using.

Reply:
Fixed with c4c8885